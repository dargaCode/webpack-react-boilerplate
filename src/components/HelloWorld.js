import React from "react";
import { join } from "lodash";

export default function HelloWorld() {
  const message = join(["Hello", "Webpack", "World!"], " ");

  return (
    <div>
      <div className="hello-world-message">{message}</div>
    </div>
  );
}
