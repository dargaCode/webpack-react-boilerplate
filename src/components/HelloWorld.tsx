import React from "react";
import { join } from "lodash";

interface Props {
  descriptor: string;
}

export default function HelloWorld(props: Props) {
  const { descriptor } = props;

  const message = join(["Hello", descriptor, "World!"], " ");

  return (
    <div>
      <div className="hello-world-message">{message}</div>
    </div>
  );
}
