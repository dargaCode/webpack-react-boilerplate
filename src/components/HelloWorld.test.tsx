import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import HelloWorld from "./HelloWorld";

describe("<HelloWorld/>", () => {
  it("should render the correct text", () => {
    const wrapper = shallow(<HelloWorld />);

    expect(wrapper.find(".hello-world-message").text()).toBe(
      "Hello Webpack World!"
    );
  });

  it("should match the snapshot", () => {
    const component = renderer.create(<HelloWorld />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
