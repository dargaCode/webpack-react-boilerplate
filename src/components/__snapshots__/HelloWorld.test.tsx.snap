// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld/> should match the snapshot 1`] = `
<div>
  <div
    className="hello-world-message"
  >
    Hello Webpack World!
  </div>
</div>
`;
